src/libs/API/parameters/UpdatePolicyConnectionConfigParams.ts

```diff
@@ -4,7 +4,7 @@ type UpdatePolicyConnectionConfigParams<TConnectionName extends ConnectionName,
     policyID: string;
     connectionName: TConnectionName;
     settingName: TSettingName;
-    settingValue?: Connections[TConnectionName]['config'][TSettingName];
+    settingValue: Connections[TConnectionName]['config'][TSettingName] | string;
```

Comment:
Revert.

In short, we can use `undefined` when making a API request but not when merging with onyx.
- The former is because we filter out undefined values [here](https://github.com/Expensify/App/blob/9d7ebf2866a144d823081bf78022096f3f8f4b65/src/libs/HttpUtils.ts#L152-L163)
- The latter is because merging undefined is nop

Suggested solution:
1. In `src/types/onyx/Policy.ts` update the types i.e. if exportEntity is optional make it optional or at least make it accept an empty string
2. In `src/libs/API/parameters/UpdatePolicyConnectionConfigParams.ts` do nothing just revert this change
3. In `src/libs/actions/connections/index.ts` replace `[settingName]: settingValue,` with  `[settingName]: settingValue ?? null,`


The last step fixes onyx migration i.e. if we call the `updatePolicyConnectionConfig` with `undefined`, onyx will use `null`



TL;DR: when clearing a value
API parameter value should be `undefined` (or empty string if that works)
Onyx value should be `null` (or empty string if that works)